avaframe/com1DFAPy/com1DFA.py

```diff
@@ -1,195 +1,1158 @@
+
+    #######################################
+    # print(np.sum(MassNearest))
+    # print(np.sum(MassBilinear))
+    # print(np.sum(MassSPH))
+
+    ###################################
+    fields['V'] = VNearest
+    fields['P'] = PNearest
+    fields['FD'] = FDNearest
+    fields['pv'] = PV
+    fields['ppr'] = PP
+    fields['pfd'] = PFD
+
+    hNN, _ = geoTrans.projectOnRasterVectRoot(x, y, FDNearest, csz=csz, interp='nearest')
+    particles['hNearestNearest'] = hNN  # np.where(h < depMin, depMin, h)
+    hNB, _ = geoTrans.projectOnRasterVectRoot(x, y, FDNearest, csz=csz, interp='bilinear')
+    particles['hNearestBilinear'] = hNB  # np.where(h < depMin, depMin, h)
+    hBN, _ = geoTrans.projectOnRasterVectRoot(x, y, FDBilinear, csz=csz, interp='nearest')
+    particles['hBilinearNearest'] = hBN  # np.where(h2 < depMin, depMin, h2)
+    hBB, _ = geoTrans.projectOnRasterVectRoot(x, y, FDBilinear, csz=csz, interp='bilinear')
+
+    # choose the interpolation method
+    indx = particles['InCell'][:, 0]
+    indy = particles['InCell'][:, 1]
+    aPart = A[indy, indx]
+    hLim = particles['m']/(rho*aPart)
+    hBB = np.where(hBB < hLim, hLim, hBB)
+    particles['hBilinearBilinear'] = hBB
+    if flagFDSPH:
+        hSPH = particles['hSPH']
+        hSPH = np.where(hSPH < hLim, hLim, hSPH)
+        particles['h'] = hSPH
+    else:
+        particles['h'] = hBB
+
+    # remove particles that have a too small height
+    # particles = removeSmallPart(hmin, particles, dem)
+
+    return particles, fields
+
+
+def plotPosition(particles, dem, data, Cmap, unit, fig, ax, plotPart=False):
+    header = dem['header']
+    ncols = header.ncols
+    nrows = header.nrows
+    xllc = header.xllcenter
+    yllc = header.yllcenter
+    csz = header.cellsize
+    xgrid = np.linspace(xllc, xllc+(ncols-1)*csz, ncols)
+    ygrid = np.linspace(yllc, yllc+(nrows-1)*csz, nrows)
+    PointsX, PointsY = np.meshgrid(xgrid, ygrid)
+    X = PointsX[0, :]
+    Y = PointsY[:, 0]
+    Z = dem['rasterData']
+    x = particles['x'] + xllc
+    y = particles['y'] + yllc
+    xx = np.arange(ncols) * csz + xllc
+    yy = np.arange(nrows) * csz + yllc
+    try:
+        # Get the images on an axis
+        cb = ax.images[-1].colorbar
+        if cb:
+            cb.remove()
+    except IndexError:
+        pass
+
+    ax.clear()
+    ax.set_title('Particles on dem at t=%.2f s' % particles['t'])
+    cmap, _, _, norm, ticks = makePalette.makeColorMap(
+        Cmap, 0.0, np.nanmax(data), continuous=contCmap)
+    cmap.set_under(color='w')
+    ref0, im = NonUnifIm(ax, xx, yy, data, 'x [m]', 'y [m]',
+                         extent=[x.min(), x.max(), y.min(), y.max()],
+                         cmap=cmap, norm=norm)
+    if plotPart:
+        ax.plot(x, y, 'ok', linestyle='None', markersize=1)
+    Cp1 = ax.contour(X, Y, Z, levels=10, colors='k')
+    addColorBar(im, ax, ticks, unit)
+    plt.pause(0.1)
+    # plt.close(fig)
```

Comment:
This is needed, right?

Reply:
the close? or the pause?

Reply:
The close

Reply:
I just update the figure so I dont want to close it